### PR TITLE
d3 heart that beats every time a message comes in

### PR DESCRIPTION
**Pull request by aloverso**
todo:
- make the messages not overlap with the heart?
- better heart graphics?
- sound

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -16,12 +16,16 @@ def get_next_message():
...
 
 	if new_message==None:
 		new_message = r.get('lastmessage')
+		resend_old_message = True
 
 	r.set('lastmessage', new_message)
-	return jsonify(json.loads(new_message.decode(encoding='UTF-8')))
+	new_message_json = json.loads(new_message.decode(encoding='UTF-8'))
+	new_message_json['resend_old_message'] = resend_old_message
+	return jsonify(new_message_json)
```
Comment (DakotaNelson) [inline server.py:28]: This returns `{'message': None, 'size': 0, 'old_message': 'another thing'}` if there's no new message?

File: templates/index.html [hunk 2]
```diff
@@ -3,19 +3,154 @@
...
+  // -----------------------------------
+
   setInterval(function() {
     $.get("/nextmessage")
-    .done(function(data) {
-      $('#subjectline').text(data['subject']);
+    .done(function(res) {
+
+      // if it's new or we just refreshed
+      // this is non-ideal (doesn't preserve list after refresh)
```
Comment (DakotaNelson) [inline templates/index.html:79]: Can we use `localStorage` or set a cookie?

File: server.py (hunk 1 above)
Comment (aloverso) [reply server.py:28]: No, if there's  no new message it pulls the last one from redis and sends it again like:
`{"message":"whatever the old message was", "size":0, "resend_old_message":true}`

Comment (DakotaNelson) [reply server.py:28]: ahhhhhhhhhh ok, seems legit

File: templates/index.html (hunk 2 above)
Comment (aloverso) [reply templates/index.html:79]: probably. don't know how to do that though :+1:

**Pull request merged**